crates/bevy_animation/src/spline_group.rs

```diff
@@ -0,0 +1,168 @@
+use splines::Spline;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum LoopStyle {
+    Once,
+    Loop,
+    PingPong,
+}
+
+pub trait SplineGroup {
+    type Sample;
+
+    fn splines(&self) -> Vec<&Spline<f32, f32>>;
+
+    fn loop_style(&self) -> LoopStyle;
+    fn loop_style_mut(&mut self) -> &mut LoopStyle;
+
+    fn time(&self) -> f32;
+    fn time_mut(&mut self) -> &mut f32;
+
+    fn speed(&self) -> f32;
+    fn speed_mut(&mut self) -> &mut f32;
+
+    fn paused(&self) -> bool;
+    fn paused_mut(&mut self) -> &mut bool;
+
+    fn pong(&self) -> bool;
+    fn pong_mut(&mut self) -> &mut bool;
+
+    fn sample(&self, time: f32) -> Self::Sample;
+
+    fn current(&self) -> Self::Sample {
+        self.sample(self.time())
+    }
+
+    fn is_empty(&self) -> bool {
+        if self.splines().is_empty() {
+            true
+        } else {
+            self.splines()
+                .into_iter()
+                .fold(true, |acc, v| if !v.is_empty() { false } else { acc })
+        }
+    }
+
+    fn start_time(&self) -> Option<f32> {
+        let starts: Vec<f32> = self
+            .splines()
+            .into_iter()
+            .map(|s| spline_start_time(s))
+            .filter_map(|s| s)
+            .collect();
+
+        if starts.is_empty() {
+            None
+        } else {
+            Some(
+                starts
+                    .iter()
+                    .fold(starts[0], |acc, v| if *v < acc { *v } else { acc }),
+            )
+        }
+    }
+
+    fn end_time(&self) -> Option<f32> {
+        let ends: Vec<f32> = self
+            .splines()
+            .into_iter()
+            .map(|s| spline_end_time(s))
+            .filter_map(|s| s)
+            .collect();
+
+        if ends.is_empty() {
+            None
+        } else {
+            Some(
+                ends.iter()
+                    .fold(ends[0], |acc, v| if *v > acc { *v } else { acc }),
+            )
+        }
+    }
+
+    fn duration(&self) -> Option<f32> {
```

Comment:
This could return `Duration` to have the time typed possibly. Otherwise I would add a comment that these are seconds.